attestation-agent/attester/src/snp/hostdata.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2024 Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::Result;
+
+pub fn get_snp_host_data() -> Result<Vec<u8>> {
+    match sev::firmware::guest::Firmware::open() {
+        Ok(mut firmware) => {
+            let report_data: [u8; 64] = [0; 64];
+            match firmware.get_report(None, Some(report_data), Some(0)) {
+                Ok(report) => Ok(report.host_data.to_vec()),
+                Err(e) => Err(e.into()),
+            }
+        }
+        Err(e) => Err(e.into()),
+    }
+}
```

Comment:
Can we use concrete error types? 

It would also be good to unnest this into:

```
{
  let firmware = sev::firmware::guest::Firmware::open()?;
  ...
  let report = firmware.get_report(...)?;
  report.host_data()
}
```

Reply:
Sounds good. Let me add the concrete error type for SNP first. Let's wait for how TDX should be implemented and then I will add for that.